modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/util/CommerceOrderItemQuantityFormatterImpl.java

```diff
@@ -148,6 +202,80 @@ private DecimalFormat _getDecimalFormat(
 		return decimalFormat;
 	}
 
+	private boolean _hasCommaDecimalPattern(String quantity) {
+		Matcher matcher = _commaDecimalPattern.matcher(quantity);
+
+		return matcher.find();
+	}
+
+	private boolean _hasPeriodDecimalPattern(String quantity) {
+		Matcher matcher = _periodDecimalPattern.matcher(quantity);
+
+		return matcher.find();
+	}
+
+	private void _validateQuantity(String className, String quantity)
+		throws Exception {
+
+		if (Validator.isNull(className) || _hasCommaDecimalPattern(quantity) ||
+			_hasPeriodDecimalPattern(quantity)) {
+
+			return;
+		}
+
+		if (Objects.equals(
```

Comment:
This is not really flexible... what if we add something else with quantities in the future?
I guess it is ok for now, but having a registry of "quantityFormatter" components identified by the className will allow us to be more flexible for extensions